CTX.js

```diff
@@ -206,49 +133,25 @@ function contextObjectXMLToCOinS (text) {
 	return titles;
 };
 
-function doImportFromText(text, showPrompt) {
-	var spans = [], items = [], zoteroItems = [];
-	
-	// This is because we want to be able to read multiple such CTX elements in a single page
-	if (typeof text != "string" && text.length >= 1) {
-		spans = text.map(contextObjectXMLToCOinS).reduce(function(a,b){return a.concat(b);});
-	} else {
-		spans = contextObjectXMLToCOinS(text);
-	}
+function doImportFromText(text) {
+	var spans = contextObjectXMLToCOinS(text);
 	
 	for (var i = 0 ; i < spans.length ; i++) {
 		Zotero.debug("Processing span: "+spans[i]);
 		var item = new Zotero.Item;
 		Zotero.Utilities.parseContextObject(spans[i], item);
-		if(item.itemType) {
+		if (item.itemType) {
 			Zotero.debug("Found " + item.itemType);
-			items.push(item.title);
-			zoteroItems.push(item);
 			// Set publicationTitle to the short title if only the latter is specified
 			if (item.journalAbbreviation && !item.publicationTitle) {
 				item.publicationTitle = item.journalAbbreviation;
 			}
-			// If we're in non-prompting mode, save right away
-			if (showPrompt === false) {
-				item.complete();
-			}
-		} else {
-			Zotero.debug("Type not found");
-		}
-	}
-	// Since we want to prompt, we have to parse twice.
-	if(showPrompt === true) {
-		if(items.length == 1) {
 			item.complete();
 		} else {
-			items = Zotero.selectItems(items);
-			if(!items) return true;
-			for(var i in items) {
-				zoteroItems[i].complete();
-			}
+			Zotero.debug("Type not found");
```

Comment:
It can have no `itemType` if Zotero doesn't recognize one (e.g. "info:ofi/fmt:kev:mtx:unknown" or something like that as returned by FreeCite), but in that case `ZU.parseContextObject` will return `false` instead of the item (we should be checking the return value here instead). The behavior of `ZU.parseContextObject` may change at some point to default to a certain item type, but `false` should always be an indicator for a failed parse.